### DISCUSSION TITLE
Mac .pkg installer signing

### DISCUSSION THREAD

**Discussion by hamada-applicita** [unoplatform/uno#15318] [Q&A] [answered]
We are using uno to build a mac catalyst application which we are distributing ourselves i.e. not through the appstore

I am using a **'Developer Id Application'** certificate and provisioning profile to sign the code and build the pkg file using this command line in my build (mac-os hosted agent azure devops)

`
 dotnet publish -f:net7.0-maccatalyst -c:Release /p:ApplicationIdentifier=com.example.app -p:CreatePackage=true -p:EnableCodeSigning=true /p:CodesignKey="Developer ID Application: Acme Ltd (xxxxxxxxx)" /p:CodesignProvision=$(provisioningProfile.provisioningProfileUuid)  /p:ArchiveOnBuild=true
 `
 
This successfully creates the .pkg file. The application installs and runs using this pkg file but macos still flags the installer as coming from an 'unidentified developer' and I need to override this warning in order to run the installer.

My understanding is that the pkg file needs to be signed with a second **'Developer Id Installer**' certificate to fix this warning. I have created this cert and am installing it to the build agent keychain and adding to the dotnet publish command

`-p:EnablePackageSigning=true`

The build then fails with 

`The identity '3rd Party Mac Developer Installer' doesn't match any valid certificate/private key pair in the default keychain`

The cert I created has key 'Developer ID Installer: AcmeLtd (xxxxxxxx)'. I need an equivalent command line option to `/p:CodesignKey` t to set this.

How to I specify the installer signing cert key? Is my understanding and approach correct to handle mac pkg creation for distribution outside of the appstore

**Comment by jeromelaban** [comment]
Thanks for the question. We do not yet have documentation for this part yet, but can use the .NET documentation of the subject. 

Here's a setup example for [the uno calculator](https://github.com/unoplatform/calculator/blob/2c130e5a6deabc068c1ff45251ffbac5b2927839/src/Calculator.Mobile/Calculator.Mobile.csproj#L119-L170) that may help you.

**Comment by hamada-applicita** [comment, marked as answer]
`PackageSigningKey ` was the setting i needed.thanks
